src/backend/distributed/utils/resource_lock.c

```diff
@@ -577,8 +577,6 @@ LockShardResource(uint64 shardId, LOCKMODE lockmode)
 	const bool sessionLock = false;
 	const bool dontWait = false;
 
-	AssertArg(shardId != INVALID_SHARD_ID);
```

Comment:
removed because this can easily be the case when using worker_append_table_to_shard

Reply:
I guess there is not much value in locking invalid_shard_id (could even be harmful). Should we return early in that case?

Reply:
There's not much downside to allowing it as far as I can see. I mainly removed it because worker_append_table_to_shard allows you to use _0 as a suffix.